kernel/device.c

```diff
@@ -50,9 +50,15 @@ void _sys_device_do_config_level(int level)
 	for (info = config_levels[level]; info < config_levels[level+1];
 								info++) {
 		struct device_config *device = info->config;
+		int res;
 
-		(void)device->init(info);
-		_k_object_init(info);
+		res = device->init(info);
+
+		if (res < 0) {
+			printk("dev init: %s: error %d\n", device->name, res);
```

Comment:
I would not use printk here, there are devices (the very early ones) for which you'll never have such output.

This error handling issue on device is not new. It goes back like 3 years ago at least, and there were no consensus (besides doing nothing that is) on how to manage it properly.

I already noted it in issue #6293

Reply:
please us LOG_ERR

Reply:
> please us LOG_ERR

So, let's compare that with what @tbursztyka said:

> there are devices (the very early ones) for which you'll never have such output.

So, my response here is: I used printk(), and not logging, here, exactly because logging may be not yet initialized there. Yes, printk may be not initialized too, but that's as good as we can do: if it's not initialized, its handler is empty, and we indeed just won't see anything. And printk() is initialized earlier than logging, or do I miss something?

Reply:
@nordic-krch : What would you say, does it make sense to use logger here (early boot)? There's no otherwise logging usage (even the older one) in this module.

Reply:
@tbursztyka :

> I already noted it in issue #6293

Ok, noted.

> This error handling issue on device is not new. It goes back like 3 years ago at least, and there were no consensus (besides doing nothing that is) on how to manage it properly.

Well, these matters are indeed not new. I complained about cryptic and obfuscated Zephyr logging couple of years ago too, and as "fix it in one motion" doesn't work, trying to do something about it since then, including not passing by this case.

So, I understand there may be no consensus whether to fail the boot entirely or just report an error and continue (like done here), but keeping not doing anything about this known issue is not an option IMHO.

Reply:
@pfalcon logger is ready to receive messages right from `_Cstart` function. It's initialized as second one after interrupt library (`_IntLibInit`).

Reply:
@nordic-krch, Thanks for confirming, will give it a try.